dissect/cstruct/cstruct.py

```diff
@@ -280,37 +314,143 @@ def resolve(self, name: str) -> BaseType:
 
         raise ResolveError(f"Recursion limit exceeded while resolving type {name}")
 
+    def _make_type(
+        self,
+        name: str,
+        bases: Iterator[object],
+        size: Optional[int],
+        *,
+        alignment: Optional[int] = None,
+        attrs: dict[str, Any] = None,
+    ) -> type[BaseType]:
+        """Create a new type class bound to this cstruct instance.
+
+        All types are created using this method. This method automatically binds the type to this cstruct instance.
+        """
+        attrs = attrs or {}
+        attrs.update(
+            {
+                "cs": self,
+                "size": size,
+                "dynamic": size is None,
+                "alignment": alignment or size,
+            }
+        )
+        return types.new_class(name, bases, {}, lambda ns: ns.update(attrs))
+
+    def _make_array(self, type_: MetaType, num_entries: Optional[int | Expression]) -> ArrayMetaType:
+        null_terminated = num_entries is None
+        dynamic = isinstance(num_entries, Expression) or type_.dynamic
+        size = None if (null_terminated or dynamic) else (num_entries * type_.size)
+        name = f"{type_.__name__}[]" if null_terminated else f"{type_.__name__}[{num_entries}]"
+
+        bases = (type_.ArrayType,)
+
+        attrs = {
+            "type": type_,
+            "num_entries": num_entries,
+            "null_terminated": null_terminated,
+            "dynamic": dynamic,
+        }
+
+        return self._make_type(name, bases, size, alignment=type_.alignment, attrs=attrs)
+
+    def _make_int_type(self, name: str, size: int, signed: bool, *, alignment: int = None) -> type[Int]:
+        return self._make_type(name, (Int,), size, alignment=alignment, attrs={"signed": signed})
+
+    def _make_packed_type(self, name: str, packchar: str, base: type, *, alignment: int = None) -> type[Packed]:
+        return self._make_type(
+            name,
+            (base, Packed),
+            struct.calcsize(packchar),
+            alignment=alignment,
+            attrs={"packchar": packchar},
+        )
+
+    def _make_enum(self, name: str, type_: MetaType, values: dict[str, int]) -> type[Enum]:
+        return Enum(self, name, type_, values)
```

Comment:
```suggestion
        return Enum(self, name, type_, **values)
```

These should be the kwargs to `EnumMetaType.__call__()`, or am I misunderstanding?

Reply:
No, this is a bit confusing but we need to be passing this dictionary as-is to Python's enum implementation. It will serve as the enum mapping.